### PR TITLE
System improvements

### PR DESCRIPTION
**Pull request by yuriy-arabskyy**
Contains following changes (mainly backtest stability improvements):
- improved tearsheet method performance by downsampling to lower timeframe if long backtest period
- added leverage and position charts to chart_signals
- fix drawdown percentage in the tearsheet, previously it was taking initial balance as reference instead of current
- added fill_at_price flag which forces the OME to execute the order at the specified price
- added reference price to signal which is set inside of the strategy context when a new signal is generated
- set strategy name as signal group name by default for strategy generated signals
- allow calling strategy fit method if at least one instrument had one update. This is to prevent strategy waiting for all symbols. Some symbols can begin years later than others.

### CODE REVIEW

File: src/qubx/backtester/simulator.py [hunk 1]
```diff
@@ -187,13 +187,16 @@ def send_order(
         price: float | None = None,
         client_id: str | None = None,
         time_in_force: str = "gtc",
+        fill_at_price: bool = False,
```
Comment (dmarienko) [inline src/qubx/backtester/simulator.py:190]: I'm thinking that this doesn't follow common interface contract for send_order if we add additional argument. May be it'd be clear if we just make interface extendable:
```
    def send_order(
        self,
        instrument: Instrument,
        order_side: str,
        order_type: str,
        amount: float,
        price: float | None = None,
        client_id: str | None = None,
        time_in_force: str = "gtc",
        **optional,                         # - all optional parameters here
    ) -> Order:
        raise NotImplementedError("send_order is not implemented")
```
and in SimulatedTrading : 
```
    def send_order(
        self,
        instrument: Instrument,
        order_side: str,
        order_type: str,
        amount: float,
        price: float | None = None,
        client_id: str | None = None,
        time_in_force: str = "gtc",
        **optional,
    ) -> Order:
        fill_at_price = optional.pop("fill_at_price", None)
        .....
```

Comment (dmarienko) [reply src/qubx/backtester/simulator.py:190]: In this case we could handle additional parameters for live orders if need (for example routing settings for interactive brokers SMART route etc etc
)

Comment (dmarienko) [reply src/qubx/backtester/simulator.py:190]: ah, I didn't see - you have already did that ))

File: src/qubx/core/context.py [hunk 2]
```diff
@@ -690,8 +706,19 @@ def trade(
...
+                fill_at_price = kwargs["fill_at_price"]
+                order_params["fill_at_price"] = fill_at_price
+                if fill_at_price:
+                    # assume worse case
+                    # TODO: add an additional flag besides price to indicate order type
+                    type = "market"
+
         order = self.trading_service.send_order(
-            instrument, side, type, size_adj, price, time_in_force=time_in_force, client_id=client_id
+            instrument, side, type, size_adj, price, time_in_force=time_in_force, client_id=client_id, **order_params
```
Comment (dmarienko) [inline src/qubx/core/context.py:721]: May be we could just pass **kwargs as is to send_order and avoid checking at 711 : 719 ?

**Review by dmarienko — COMMENTED** [review]
Looks good - please check my comment in contex.py

File: src/qubx/backtester/simulator.py (hunk 1 above)
Comment (yuriy-arabskyy) [reply src/qubx/backtester/simulator.py:190]: but I agree it is cleaner to pass **optional and unpack inside of the send_order

File: src/qubx/core/context.py (hunk 2 above)
Comment (yuriy-arabskyy) [reply src/qubx/core/context.py:721]: yes-yes, makes sense

**Pull request merged**